src/server.ts

```diff
@@ -1,97 +1,246 @@
+    const size: number = parseInt(call.request.size, 10);
+    return FileService.createUpload(key, bucket, name, ownerID, parent, size);
+  }
+
+  // Updates the uploadID.
+  private async UpdateUploadID(call: any, callback: any): Promise<IUpload> {
+    const key: string = call.request.key;
+    const uploadID: string = call.request.uploadID;
+    const bucket: string = call.request.bucket;
+    return FileService.updateUpload(uploadID, key, bucket);
+  }
+
+  // Get an upload metadata by its id in the DB.
+  private async GetUploadByID(call: any, callback: any): Promise<IUpload> {
+    const id: string = call.request.uploadID;
+    return FileService.getUploadById(id);
+  }
+
+  //  Delete an upload from the DB by its id.
+  private async DeleteUploadByID(call: any, callback: any): Promise<void> {
+    const id: string = call.request.uploadID;
+    return FileService.deleteUpload(id);
+  }
+
+  // ********************* FILE FUNCTIONS ********************* */
+
+  // Creates a new file in the DB.
+  private async CreateFile(call: any, callback: any): Promise<ResFile> {
+    const params = call.request;
+    const createdFile = await FileService.create(
+      params.bucket,
+      params.name,
+      params.ownerID,
+      params.type,
+      params.parent,
+      params.key,
+      parseInt(params.size, 10),
+    );
+    return new ResFile(createdFile);
+  }
+
+  // Deletes a file, according to the file deletion policy.
+  private async DeleteFile(call: any, callback: any): Promise<{ok: boolean}> {
+    const id: string = call.request.id;
+    await FileService.delete(id);
+    return { ok: true };
+  }
+
+  // Retrieves a file by its id.
+  private async GetFileByID(call: any, callback: any): Promise<ResFile> {
+    const id: string = call.request.id;
+    const file: IFile = await FileService.getById(id);
+    return new ResFile(file);
+  }
+
+  // Retrieves a file by its key.
+  private async GetFileByKey(call: any, callback: any): Promise<ResFile> {
+    const key: string = call.request.key;
+    const file: IFile = await FileService.getByKey(key);
+    return new ResFile(file);
   }
+
+  // Retrieves all files residing in a given folder.
+  private async GetFilesByFolder(call: any, callback: any): Promise<{ files: ResFile[] }> {
+    const folderID: string = call.request.folderID;
+    const ownerID: string = call.request.ownerID;
+    const files: IFile[] = await FileService.getFilesByFolder(folderID, ownerID);
+    const resFiles: ResFile[] = files.length ? files.map(file => new ResFile(file)) : [];
+    return { files: resFiles };
+  }
+
+  // Checks if an operation is allowed by permission of the owner.
+  private async IsAllowed(call: any, callback: any): Promise<{ allowed: boolean }> {
+    const res: boolean = await FileService.isOwner(call.request.fileID, call.request.userID);
+    return  { allowed: res };
+  }
+
 }
 
-if (!module.parent) {
-  new Server().app;
+// Same as IFile, but changing types accordingly
```

Comment:
?

Reply:
moved to file.interface. 
the types cause a bit of problem because of the dates. we can talk about it tomorrow morning